saleor/graphql/checkout/utils.py

```diff
@@ -0,0 +1,70 @@
+from enum import Enum
+from typing import Iterable
+
+from django.core.exceptions import ValidationError
+
+from ...checkout.models import Checkout, CheckoutLine
+from ...checkout.utils import is_fully_paid, is_valid_shipping_method
+from ...discount import DiscountInfo
+
+
+def clean_checkout_shipping(
+    checkout: Checkout,
+    lines: Iterable[CheckoutLine],
+    discounts: Iterable[DiscountInfo],
+    error_code: Enum,
```

Comment:
Owczar suggested that this could be union of `PaymentErrorCode` and `CheckoutErrorCode`.